gateway/gateway_test.go

```diff
@@ -544,3 +547,40 @@ func TestGoGetSupport(t *testing.T) {
 	assert.Nil(t, err)
 	assert.Equal(t, http.StatusOK, res.StatusCode)
 }
+
+func TestTrustlessMode(t *testing.T) {
```

Comment:
Only tested for `/ipfs`. Not sure how we can simulate `/ipns` here with keys. We can only simulate DNSLink the way we did in the other tests. This is something we should figure out at some point.